File: webapp/src/main/java/io/ehdev/conrad/app/service/repo/model/RepoResponseModel.java
package io.ehdev.conrad.app.service.repo.model;

import com.fasterxml.jackson.annotation.JsonView;
import io.ehdev.conrad.database.model.VcsRepoModel;

public class RepoResponseModel extends RepoCreateModel {

    @JsonView(ViewPermission.Public.class)
    private String id;

    @JsonView(ViewPermission.Private.class)
    private String token;

    public RepoResponseModel(String name, String url, String bumper, String id, String token) {
        super(name, url, bumper);
        this.id = id;
        this.token = token;
    }

    public RepoResponseModel() {
    }

    public RepoResponseModel(VcsRepoModel repo) {
        this(repo.getRepoName(), repo.getUrl(), repo.getVersionBumperName(), repo.getUuid(), repo.getToken());
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }
}

File: api-model/src/main/kotlin/io/ehdev/conrad/model/repo/RepoResponseModel.kt
package io.ehdev.conrad.model.repo

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonView

class RepoResponseModel : AbstractRepoCreateModel {

    @JsonView(RepoView.Public::class)
    var id: String;

    @JsonView(RepoView.Private::class)
    var token: String;

    @JsonCreator
    constructor(@JsonProperty("name") name: String,
                @JsonProperty("url") url: String?,
                @JsonProperty("bumper") bumper: String,
                @JsonProperty("id") id: String,
                @JsonProperty("token") token: String) : super(name, url, bumper) {
        this.id = id
        this.token = token
    }
}
